### PR TITLE
[Requires all dev approval] Updates dev dependancies

### PR DESCRIPTION
**Pull request by Moejoe90**
**Important updates (please read):** 
Typescript:
- https://devblogs.microsoft.com/typescript/announcing-typescript-3-4/
- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-3.html

Updates dev dependencies
however, it doesn't touch typing much as they are more dependent on our regular dependencies.

`cypress` with the latest update started to declare chai and jquery as global namespaces this is creating a mess in all the other apps
the solution was to patch the cypress package and manually remove their typing so that they can relay on the already declared typings from `@type/chai` and `@type/jquery`

Also  `jsdom` `fetch-mock` `ts-node` `css-loader` were reverted as the api changed in a significant way and require more time to solve the problems that are showing up

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,53 +110,53 @@
...
-    "cypress": "^3.1.5",
-    "danger": "^3.8.6",
-    "dotenv": "^6.0.0",
-    "enzyme": "^3.7.0",
+    "clean-webpack-plugin": "^2.0.1",
+    "codecov": "^3.2.0",
+    "concurrently": "^4.1.0",
+    "copy-webpack-plugin": "^5.0.2",
+    "css-loader": "1.0.1",
+    "cypress": "^3.2.0",
```
Comment (desfero) [inline package.json]: can you please rever cypress changes, i will push this as a separate PR soon

File: package.json [hunk 2]
```diff
@@ -110,53 +110,53 @@
...
-    "css-loader": "^1.0.1",
-    "cypress": "^3.1.5",
-    "danger": "^3.8.6",
-    "dotenv": "^6.0.0",
-    "enzyme": "^3.7.0",
+    "clean-webpack-plugin": "^2.0.1",
+    "codecov": "^3.2.0",
+    "concurrently": "^4.1.0",
+    "copy-webpack-plugin": "^5.0.2",
+    "css-loader": "1.0.1",
```
Comment (desfero) [inline package.json]: what's the reason to remove `^`?

File: package.json [hunk 3]
```diff
@@ -110,53 +110,53 @@
...
+    "cypress": "^3.2.0",
+    "danger": "^7.0.17",
+    "dotenv": "6.0.0",
```
Comment (desfero) [inline package.json]: what's the reason to remove ^?

File: package.json [hunk 4]
```diff
@@ -110,53 +110,53 @@
...
+    "enzyme": "^3.9.0",
     "enzyme-adapter-react-16": "^1.7.0",
-    "express": "^4.16.3",
+    "express": "^4.16.4",
     "express-history-api-fallback": "^2.2.1",
-    "fetch-mock": "^6.0.0",
-    "file-loader": "^2.0.0",
+    "fetch-mock": "6.0.0",
```
Comment (desfero) [inline package.json]: what's the reason to remove ^?

File: package.json [hunk 5]
```diff
@@ -110,53 +110,53 @@
...
+    "file-loader": "^3.0.1",
     "font-awesome": "^4.7.0",
-    "fork-ts-checker-webpack-plugin": "^0.5.1",
-    "ganache-cli": "^6.1.7",
-    "glob": "^7.1.2",
+    "fork-ts-checker-webpack-plugin": "^1.0.1",
+    "ganache-cli": "^6.4.1",
```
Comment (desfero) [inline package.json]: btw, do we still need ganache? It would be really hard to run now node locally

File: package.json [hunk 6]
```diff
@@ -110,53 +110,53 @@
...
+    "glob": "^7.1.3",
     "html-webpack-plugin": "^3.2.0",
-    "http-proxy-middleware": "^0.18.0",
+    "http-proxy-middleware": "^0.19.1",
     "ignore-styles": "^5.0.1",
-    "jsdom": "^11.5.1",
+    "jsdom": "11.11.0",
```
Comment (desfero) [inline package.json]: what's the reason to remove ^?

File: package.json [hunk 7]
```diff
@@ -166,31 +166,31 @@
...
+    "ts-node": "6.2.0",
+    "tslint": "^5.14.0",
     "tslint-eslint-rules": "^5.4.0",
-    "typechain": "^0.2.7",
+    "typechain": "0.2.7",
     "types-bn": "^0.0.1",
     "types-ethereumjs-util": "^0.0.5",
-    "typescript": "^3.2.2",
-    "typestrict": "^1.0.1",
+    "typescript": "^3.3.4000",
```
Comment (desfero) [inline package.json]: can you update this to the 3.4.1?

File: package.json [hunk 8]
```diff
@@ -166,31 +166,31 @@
...
-    "ts-loader": "^5.3.1",
-    "ts-node": "^6.2.0",
-    "tslint": "^5.13.1",
+    "terser-webpack-plugin": "^1.2.3",
+    "ts-loader": "^5.3.3",
+    "ts-node": "6.2.0",
+    "tslint": "^5.14.0",
     "tslint-eslint-rules": "^5.4.0",
-    "typechain": "^0.2.7",
+    "typechain": "0.2.7",
```
Comment (desfero) [inline package.json]: what's the reason to remove ^?

File: package.json [hunk 9]
```diff
@@ -166,31 +166,31 @@
...
+    "typescript": "^3.3.4000",
+    "typestrict": "^1.0.2",
     "url-loader": "^1.1.2",
     "web3-eth-accounts": "^1.0.0-beta.36",
-    "web3-typescript-typings": "^0.9.3",
-    "webpack": "^4.29.0",
-    "webpack-bundle-analyzer": "^3.0.3",
-    "webpack-cli": "^3.1.2",
-    "webpack-dev-server": "^3.1.10",
+    "web3-typescript-typings": "0.9.3",
```
Comment (desfero) [inline package.json]: what's the reason to remove ^?

File: package.json (hunk 6 above)
Comment (Moejoe90) [reply package.json]: The version after this is introducing breaking changes. I didn't want to deal with these breaking changes in this PR

File: app/types.d.ts [hunk 10]
```diff
@@ -17,7 +17,9 @@ export type FunctionWithDeps = Opaque<"FunctionWithDeps", Function>;
 type DeepPartial<T> = {
   [P in keyof T]?: T[P] extends Array<infer U>
     ? Array<DeepPartial<U>>
-    : T[P] extends ReadonlyArray<infer U> ? ReadonlyArray<DeepPartial<U>> : DeepPartial<T[P]>
+    : T[P] extends ReadonlyArray<infer U>
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/types.d.ts]: Maybe the `prettier` guys decided that like this its `prettier`

File: patches/bitcore-mnemonic+1.7.0.patch [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
+index 513eb36..dba006d 100644
 --- a/node_modules/bitcore-mnemonic/lib/mnemonic.js
 +++ b/node_modules/bitcore-mnemonic/lib/mnemonic.js
 @@ -94,7 +94,7 @@ var Mnemonic = function(data, wordlist) {
    });
  };
  
 -Mnemonic.Words = require('./words');
-+Mnemonic.Words["ENGLISH"] = require('./words').ENGLISH;
++Mnemonic.Words = {"ENGLISH": require('./words').ENGLISH};
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply patches/bitcore-mnemonic+1.7.0.patch]: I was getting errors that `Mnemonic.Words` was not accessible so i had to change it in a way that initializes  `Mnemonic.Words` without assuming that `Words` exists

File: package.json (hunk 1 above)
Comment (Moejoe90) [reply package.json]: :+1:

File: package.json (hunk 5 above)
Comment (Moejoe90) [reply package.json]: True we can take it off now

**Pull request merged**